web/modules/custom/weather/src/Plugin/Block/WeatherBlock.php

```diff
@@ -151,4 +218,76 @@ private function getLocation() {
     return $this->config->get('weather.default_city') ?? 'Lutsk, Ukraine';
   }
 
+  /**
+   * Save Cookie with location, help to avoid many unnecessary requests to API.
+   *
+   * @param string $location
+   *   A location data for save it in cookie.
+   */
+  private function setCookie(string $location) {
+    user_cookie_save(['location' => $location]);
+  }
+
+  /**
+   * Get the user location if it appears in cookies.
+   *
+   * @return string|null
+   *   Return a data from Cookies if exists a 'location' key
+   *   when key not appears in Cookies - return NULL.
+   */
+  private function getCookie() {
+    return $this->stack->getCurrentRequest()->cookies->get('Drupal_visitor_location') ?? NULL;
```

Comment:
1) Where is the cookie 'Drupal_visitor_location' saved?
2) It seems like the saved location cannot be changed. But I as a user can open your site from Kovel then go to Lutsk and open the site again. Did you handle this case?